packages/common-integration-tests/src/auth-client.ts

```diff
@@ -336,6 +351,207 @@ export function runAuthClientTests(
+        new Permissions([
+          new CachePermission(CacheRole.ReadWrite, {
+            cache: {name: FGA_CACHE_1},
+          }),
+          new TopicPermission(TopicRole.ReadWrite, {
+            cache: {name: FGA_CACHE_1},
+          }),
+        ]),
+        ExpiresIn.seconds(10)
+      );
+      expect(tokenResponse).toBeInstanceOf(GenerateAuthToken.Success);
+      const token = (tokenResponse as GenerateAuthToken.Success).authToken;
+      const cacheClient = cacheClientFactory(token);
+
+      // Read/Write on cache `foo` is allowed
+      const setResp = await cacheClient.set(FGA_CACHE_1, 'ned', 'flanders');
+      expect(setResp).toBeInstanceOf(CacheSet.Success);
+
+      const hitResp = await cacheClient.get(FGA_CACHE_1, 'ned');
+      expect(hitResp).toBeInstanceOf(CacheGet.Hit);
+      expect(hitResp).toHaveProperty('valueString', 'flanders');
+
+      // Read/Write on cache `bar` is not allowed
+      const setResp1 = await cacheClient.set(FGA_CACHE_2, 'flaming', 'mo');
+      expect(setResp1).toBeInstanceOf(CacheSet.Error);
+      expect(setResp1).toHaveProperty(
+        'errorCode',
+        MomentoErrorCode.PERMISSION_ERROR
+      );
+      const getResp1 = await cacheClient.get(FGA_CACHE_2, 'flaming');
+      expect(getResp1).toBeInstanceOf(CacheGet.Error);
+      expect(getResp1).toHaveProperty(
+        'errorCode',
+        MomentoErrorCode.PERMISSION_ERROR
+      );
+
+      const topicClient = topicClientFactory(token);
+      // Read/Write on topics in cache foo not allowed
+      const pubResp = await topicClient.publish(
+        FGA_CACHE_1,
+        FGA_TOPIC_1,
+        'breaking news!'
+      );
+      expect(pubResp).toBeInstanceOf(TopicPublish.Error);
+      expect(pubResp).toHaveProperty(
+        'errorCode',
+        MomentoErrorCode.PERMISSION_ERROR
+      );
+      const subResp = await topicClient.subscribe(
+        FGA_CACHE_1,
+        FGA_TOPIC_1,
+        trivialHandlers
+      );
+      expect(subResp).toBeInstanceOf(TopicSubscribe.Error);
+      expect(subResp).toHaveProperty(
+        'errorCode',
+        MomentoErrorCode.PERMISSION_ERROR
+      );
+
+      // Read/Write on topics in cache bar is allowed
+      expect(
+        await topicClient.publish(FGA_CACHE_2, FGA_TOPIC_2, 'breaking news!')
+      ).toBeInstanceOf(TopicPublish.Success);
+      const subscribeResponse = await topicClient.subscribe(
+        FGA_CACHE_2,
+        FGA_TOPIC_2,
+        trivialHandlers
+      );
+      expectWithMessage(() => {
+        expect(subscribeResponse).toBeInstanceOf(TopicSubscribe.Subscription);
+      }, `expected SUBSCRIPTION but got ${subscribeResponse.toString()}`);
+    });
+
+    afterAll(async () => {
+      expect(
+        await sessionTokenCacheClient.deleteCache(FGA_CACHE_1)
+      ).toBeInstanceOf(DeleteCache.Success);
+      expect(
+        await sessionTokenCacheClient.deleteCache(FGA_CACHE_2)
+      ).toBeInstanceOf(DeleteCache.Success);
+    });
+  });
```

Comment:
would love to see one more test that did ReadOnly on topics

Reply:
Added in https://github.com/momentohq/client-sdk-javascript/pull/598/commits/f2aad35500cbec9f6c42ca858e70abafffacbf58